### PR TITLE
python_ot: update to match upstream OpenTitan

### PR DESCRIPTION
**Pull request by nbdd0121**
This updates python_ot environment to match current OpenTitan master branch.
* Git dependency is switched to URL dependency
* FuseSoC is updated

### CODE REVIEW

File: pkgs/python_ot/pyproject.toml [hunk 1]
```diff
@@ -95,9 +95,9 @@ wcwidth = "0.2.6"
 wheel = "0.41.2"
 yapf = "0.32.0"
 zipfile2 = "0.0.12"
-chipwhisperer = { git = "https://github.com/newaetech/chipwhisperer-minimal.git", rev = "2643131b71e528791446ee1bab7359120288f4ab" }
-edalize = { git = "https://github.com/lowRISC/edalize.git", tag = "v0.4.0"}
-fusesoc = { git = "https://github.com/lowRISC/fusesoc.git", tag = "ot-0.4" }
+chipwhisperer = { url = "https://github.com/newaetech/chipwhisperer-minimal/archive/2643131b71e528791446ee1bab7359120288f4ab.zip" }
+edalize = { url = "https://github.com/lowRISC/edalize/archive/refs/tags/v0.4.0.zip" }
+fusesoc = { url = "https://github.com/lowRISC/fusesoc/archive/refs/tags/ot-0.5.zip" }
```
Comment (HU90m) [inline pkgs/python_ot/pyproject.toml:100]: Is `url` recommended over `git`?

File: pkgs/python_ot/default.nix [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
+  # This overlay is generated by poetry2nix
+  poetry-git-overlay = import ./poetry-git-overlay.nix {inherit pkgs;};
+
   poetry2nix = inputs.poetry2nix.lib.mkPoetry2Nix {inherit pkgs;};
   poetryOverrides = inputs.self.lib.poetryOverrides {inherit pkgs;};
 in
   poetry2nix.mkPoetryEnv {
     projectDir = ./.;
     overrides = [
+      poetry-git-overlay
```
Comment (HU90m) [inline pkgs/python_ot/default.nix:19]: What's this file do?

Comment (nbdd0121) [reply pkgs/python_ot/default.nix:19]: This is generated by poetry2nix for pinning URL dependencies

File: pkgs/python_ot/pyproject.toml (hunk 1 above)
Comment (nbdd0121) [reply pkgs/python_ot/pyproject.toml:100]: pip can't pin git dependencies with hash, so OT currently use url dependency.

**Review by hcallahan-lowrisc — APPROVED** [review]

**Pull request merged**